File: fhc_2014/QR_B.py
# Facebook Hacker Cup 2014 Qualification Round
# B. Basketball Game

import sys

def solve():
  (N, M, P) = [int(x) for x in sys.stdin.readline().rstrip().split(' ')]
  m = []
  for i in range(N):
    (name, shot, height) = sys.stdin.readline().rstrip().split(' ')
    shot = int(shot)
    height = int(height)
    m.append([0, 0, -shot, -height, name])
  m = sorted(m)
  for k,v in enumerate(m):
    m[k][1] = k
  all = [m[::2], m[1::2]]
  active = [all[0][:P], all[1][:P]]
  bench = [all[0][P:], all[1][P:]]
  for min in range(M):
    for i in range(2):
      for k,v in enumerate(active[i]):
        active[i][k][0] += 1
      if len(bench[i]) > 0:
        a = active[i][:-1] + [bench[i][0]]
        b = bench[i][1:] + [active[i][-1]]
        active[i] = sorted(a)
        bench[i] = sorted(b)
  return [x[-1] for x in active[0]+active[1]]

cases = int(sys.stdin.readline())
for case in range(cases):
  r = solve()
  s = "Case #" + str(case + 1) + ": "
  s += " ".join(sorted(r))
  print(s)

File: fhc_2020/QR_B.cpp
//
// Facebook Hacker Cup 2020 Qualification Round
// B. Alchemy
//

#include <iostream>
#include <sstream>
#include <map>

using namespace std;

bool solve(int n, string s) {
	string x;
	for (char c : s) {
		x += c;
		while (x.length() >= 3) {
			map<int, int> m;
			m[x.back()] += 1;
			m[x[x.length() - 2]] += 1;
			m[x[x.length() - 3]] += 1;
			if (m['A'] == 2) {
				x.resize(x.length() - 3);
				x += 'A';
			} else if (m['B'] == 2) {
				x.resize(x.length() - 3);
				x += 'B';
			} else {
				break;
			}
		}
	}
	return x.length() == 1;
}

int main(int argc, char* argv[]) {
	int T, n;
	cin >> T;
	for (int t = 1; t <= T; ++t) {
		cin >> n;
		string s;
		cin >> s;
		bool ans = solve(n, s);
		cout << "Case #" << t << ": " << (ans ? "Y" : "N") << endl;
	}
	return 0;
}
